File: WordMaster.Domain/Entities/LogHistory.cs
namespace WordMaster.Domain.Entities;

public class LogHistory
{
    public long Id { get; set; }
    public Guid? AppUserId { get; set; }
    public string? Email { get; set; }
    public string? IpAddress { get; set; }
    public bool IsSuccessful { get; set; }
    public DateTime AttemptedAt { get; set; } = DateTime.UtcNow;
    public string? Source { get; set; }
    public AppUser? AppUser { get; set; }
}

File: frontend/src/types/logHistory.ts
import type { PagedResult } from "./api";

export interface LogHistoryResponse {
    id: number;
    appUserId?: string;
    email?: string;
    ipAddress?: string;
    isSuccessful: boolean;
    attemptedAt: string;
    source?: string;
}

export interface GetLogHistoryRequest {
    page: number;
    size: number;
    searchTerm?: string;
    searchInEmail: boolean;
    searchInUserId: boolean;
    searchInIp: boolean;
}

export interface LogHistoryListResponse extends PagedResult<LogHistoryResponse> { }
